lib/gourami/coercer.rb

```diff
@@ -120,21 +119,38 @@ def coerce_phone(value, options = {})
     #   The type of the hash keys to coerce, no coersion if value is nil.
     # @option options :value_type [Symbol, Callable] (nil)
     #   The type of the hash values to coerce, no coersion if value is nil.
+    # @option options :indifferent_access [Boolean] (false)
+    #   When true, the resulting Hash will be an ActiveSupport::HashWithIndifferentAccess
     #
-    # @return [Hash]
+    # @return [Hash, ActiveSupport::HashWithIndifferentAccess]
     #   The coerced Hash.
     def coerce_hash(value, options = {})
+      return if options[:allow_nil] && value.nil?
+
       hash_key_type = options[:key_type]
       hash_value_type = options[:value_type]
 
-      return {} unless value.is_a?(Hash) || (defined?(Sequel::Postgres::JSONHash) && value.is_a?(Sequel::Postgres::JSONHash))
+      hash_class = options[:indifferent_access] ? ActiveSupport::HashWithIndifferentAccess : Hash
+      hash = hash_class.new
+
+      return hash unless value.is_a?(Hash) || (defined?(Sequel::Postgres::JSONHash) && value.is_a?(Sequel::Postgres::JSONHash))
```

Comment:
On this line should we also check if `value.is_a?(ActiveSupport::HashWithIndifferentAccess)` or `ActiveSupport::HashWithIndifferentAccess` is basically a `Hash`?

Reply:
```
[3] Vydia(main)> ActiveSupport::HashWithIndifferentAccess.new.is_a?(Hash)
=> true
```